packages/main/src/AvatarGroup.js

```diff
@@ -0,0 +1,456 @@
+			detail: {
+				targetRef: { type: HTMLElement },
+				overflowButtonClicked: { type: Boolean },
+			},
+		},
+	},
+};
+
+/**
+ * @class
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ *
+ * Displays a group of avatars arranged horizontally. It is useful to visually
+ * showcase a group of related avatars, such as, project team members or employees.
+ *
+ * The control allows you to display the avatars in different sizes,
+ * depending on your use case.
+ *
+ * The <code>AvatarGroup</code> control has two group types:
+ * <ul>
+ * <li><code>Group</code> type: The avatars are displayed as partially overlapped on
+ * top of each other and the entire group has one click/tap area.</li>
+ * <li><code>Individual</code> type: The avatars are displayed side-by-side and each
+ * avatar has its own click/tap area.</li>
+ * </ul>
+ *
+ * <h3>Responsive Behavior</h3>
+ *
+ * When the available space is less than the width required to display all avatars,
+ * an overflow visualization appears as a button placed at the end with the same shape
+ * and size as the avatars. The visualization displays the number of avatars that have overflowed
+ * and are not currently visible.
+ *
+ * <h3>Usage</h3>
+ *
+ * Use the <code>AvatarGroup</code> if:
+ * <ul>
+ * <li>You want to display a group of avatars.</li>
+ * <li>You want to display several avatars which have something in common.</li>
+ * </ul>
+ *
+ * Do not use the <code>AvatarGroup</code> if:
+ * <ul>
+ * <li>You want to display a single avatar.</li>
+ * <li>You want to display a gallery for simple images.</li>
+ * <li>You want to use it for other visual content than avatars.</li>
+ * </ul>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.main.AvatarGroup
+ * @extends UI5Element
+ * @tagname ui5-avatar-group
+ * @since 1.0.0-rc.11
+ * @public
+ */
+class AvatarGroup extends UI5Element {
+	constructor() {
+		super();
+		this._itemNavigation = new ItemNavigation(this);
+
+		this._colorIndex = 0;
+		this._hiddenItems = 0;
+		this._onResizeHandler = this._onResize.bind(this);
+	}
+
+	static get metadata() {
+		return metadata;
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get template() {
+		return AvatarGroupTemplate;
+	}
+
+	static get styles() {
+		return AvatarGroupCss;
+	}
+
+	/**
+	 * Returns array with hidden items. Items that are in overflow button.
+	 * @readonly
+	 * @type { Array }
+	 * @defaultValue []
+	 * @public
+	 */
+	get hiddenItems() {
+		return this.items.slice(this._hiddenStartIndex);
+	}
+
+	get _hiddenStartIndex() {
+		return this._itemsCount - this._hiddenItems;
+	}
+
+	get _overflowBtnHidden() {
+		return this._hiddenItems === 0;
+	}
+
+	get _isGroup() {
+		return this.type === AvatarGroupType.Group;
+	}
+
+	get _itemsCount() {
+		return this.items.length;
+	}
+
+	get _groupTabIndex() {
+		return this._isGroup ? "0" : "-1";
+	}
+
+	get _overflowButtonTabIndex() {
+		return this._isGroup ? "-1" : false;
+	}
+
+	get _colorBackgroundIndex() {
```

Comment:
Should be `_backgroundColorIndex` because you mean "the index of the background color". The way you named it means "the color's background index"